index.bs

```diff
@@ -947,42 +923,27 @@ spec: RFC7240; urlPrefix: https://tools.ietf.org/html/rfc7240
     <li>
       Let <var>upgrade state</var> be the result of executing
       [[#should-upgrade-for-client]] upon <var>request</var>'s
-      <a for="request" attribute>client</a>.
+      <a for="request">client</a>.
     </li>
     <li>
       If <var>upgrade state</var> is <a value>Do Not Upgrade</a>, return without
       modifying <var>request</var>.
     </li>
     <li>
-      If <var>request</var>'s <a for="request" attribute>url</a>'s {{URL/scheme}} is
-      <code>http</code>, set <var>request</var>'s <a for="request" attribute>url</a>'s
-      {{URL/scheme}} to <code>https</code>, and return.
-    </li>
-    <li>
-      If <var>request</var>'s <a for="request" attribute>url</a>s {{URL/port}} is
-      <code>80</code>, set <var>request</var>'s <a for="request" attribute>url</a>s {{URL/port}}
-      to <code>443</code>.
-
-      Note: This will only change the URL's port if the port is
-      explicitly set to <code>80</code>. If the port is not set, or if
-      it is set to some non-standard value, the user agent will not
-      modify that value. This implementation makes the same tradeoffs as
-      HSTS (see [[RFC6797]], and specifically step #5 of
-      <a href="https://tools.ietf.org/html/rfc6797#section-8.3">Section
-      8.3</a>, and item #6 in
-      <a href="https://tools.ietf.org/html/rfc6797#appendix-A">Appendix
-      A</a>).
+      If <var>request</var>'s <a for="request">URL</a>'s <a for=url>scheme</a> is
+      "<code>http</code>", set <var>request</var>'s <a for="request">URL</a>'s
+      <a for=url>scheme</a> to "<code>https</code>", and return.
```

Comment:
It might be worth keeping something like the note to make it clear to developers what expectations they ought to have for non-standard ports. I don't think you need to do that in this patch, though. I can add something to the "authoring considerations" section in a subsequent PR.